contracts/Oracles/RedstoneConsumerWrapper.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { ProxyConnector } from "@redstone-finance/evm-connector/contracts/core/ProxyConnector.sol";
+import { RedstoneDefaultsLib } from "@redstone-finance/evm-connector/contracts/core/RedstoneDefaultsLib.sol";
+import { IRedstoneConsumer, IRedstoneConsumerWrapper } from "./Interfaces/IRedstoneConsumerWrapper.sol";
+
+contract RedstoneConsumerWrapper is IRedstoneConsumerWrapper, ProxyConnector {
+    // --- Immutable variables ---
+
+    IRedstoneConsumer public immutable override redstoneConsumer;
+
+    // --- Variables ---
+
+    uint256 public override lastPrice;
+    uint256 public override lastUpdateTimestamp;
+    bool public override isBrokenOrFrozen;
+
+    // --- Constructor ---
+    constructor(IRedstoneConsumer _redstoneConsumer) {
+        if (address(_redstoneConsumer) == address(0)) {
+            revert RedstoneConsumerCannotBeZeroAddress();
+        }
+        redstoneConsumer = _redstoneConsumer;
+    }
+
+    // --- Functions ---
+
+    function setPrice(bytes calldata redstonePayload) external {
+        try redstoneConsumer.getPrice(redstonePayload) returns (uint256 oraclePrice) {
+            lastPrice = oraclePrice;
+            lastUpdateTimestamp = block.timestamp;
+            isBrokenOrFrozen = false;
```

Comment:
Can we remove this? when consuming we will check for timestamp anyway. In the case of setting isBrokenOrFrozen  to true, we do not set timestamp anyway (it would not meet the condition that price is from this block).

Reply:
Probably leave as is for now, then we can think of fixing it.
In order for this optimization to be safe, we would need to put `onlyRedStonePositionsManager` or something like that

Reply:
I think that 'onlyRedStonePositionsManager' is unnecessary because this function calls `redstoneConsumer.getPrice(redstonePayload)`, which is our contract too.